ecommerce/backend/API/views/recommendation.py

```diff
@@ -0,0 +1,81 @@
+from rest_framework import exceptions
+from rest_framework.response import Response
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework import status
+import json
+import random
+
+from ..utils import permissions, Role
+from ..models import Product,Vendor,ImageUrls,Category,Subcategory,Purchase, Order, User
+from ..serializers.address_serializer import AddressResponseSerializer
+from ..serializers.product_serializer import ProductResponseSerializer
+from ..serializers.purchase_serializer import PurchaseResponseSerializer
+from ..utils.order_status import OrderStatus
+
+@api_view(['GET'])
+@permission_classes([permissions.IsCustomerUser])
+def recommend_products(request):
+
+    # get user
+    user = request.user
+    user_id = request.user.id
+
```

Comment:
Here if no token is provided in request header, and user will be none. You should check for that and return an error in case